### PR TITLE
Feature: Add mtREPLAY_DELTA_RESPONSE message test.

### PR DESCRIPTION
**Pull request by lwojtow**
1. Add required variants for Payload enum.
2. Add test for mtREPLAY_DELTA_RESPONSE message.

### CODE REVIEW

File: SPEC.md [hunk 1]
```diff
@@ -268,3 +268,11 @@ The test index makes use of symbolic language in describing connection and messa
 
     <>
     <- mtVALIDATION
+
+### ZG-CONFORMANCE-022
+
+    The node should respond with mtREPLAY_DELTA_RESPONSE for mtREPLAY_DELTA_REQ.
```
Comment (ljedrz) [inline SPEC.md:274]:
```suggestion
    The node should respond with mtREPLAY_DELTA_RESPONSE to mtREPLAY_DELTA_REQ.
```

File: src/tests/conformance/query/replay_delta.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    let ledger_hash =
+        hex::decode(ledger_info.result.ledger.ledger_hash).expect("Unable to decode ledger hash.");
+    let payload = Payload::TmReplayDeltaRequest(TmReplayDeltaRequest {
+        ledger_hash: ledger_hash.clone(),
+    });
+
+    // Send a message from the synthetic node.
+    synth_node
+        .unicast(node.addr(), payload)
+        .expect("Unable to send message.");
```
Comment (ljedrz) [inline src/tests/conformance/query/replay_delta.rs:43]:
```suggestion
        .expect("Unable to send a message.");
```

File: src/protocol/handshake.rs [hunk 3]
```diff
@@ -112,7 +112,7 @@ impl Handshake for InnerNode {
                 request.extend_from_slice(b"Connect-As: Peer\r\n");
                 request.extend_from_slice(format!("Public-Key: {}\r\n", base58_pk).as_bytes());
                 request.extend_from_slice(format!("Session-Signature: {}\r\n", sig).as_bytes());
-                request.extend_from_slice(b"X-Protocol-Ctl: txrr=1\r\n");
+                request.extend_from_slice(b"X-Protocol-Ctl: txrr=1;ledgerreplay=1\r\n");
```
Comment (ljedrz) [inline src/protocol/handshake.rs:115]: what's the rationale behind this? we should probably document it

**Review by ljedrz — APPROVED** [review]
LGTM with an extra doc request

File: SPEC.md (hunk 1 above)
Comment (lwojtow) [reply SPEC.md]: Done

File: src/protocol/handshake.rs (hunk 3 above)
Comment (lwojtow) [reply src/protocol/handshake.rs:117]: Added a short comment here and a longer explanation in the SPEC file.

**Review by ljedrz — APPROVED** [review]

**Review by Rqnsom — COMMENTED** [review]
Here's one more "LGTM" 💯 

Do we usually want 2/2 approvals?

**Review by Rqnsom — APPROVED** [review]

**Comment by ljedrz** [comment]
> Do we usually want 2/2 approvals?

I'd say this depends on the level of complexity, judged by either the submitter or the first reviewer.

**Pull request merged**